### PR TITLE
Doafter cleanup, disables stoplag dynamic sleeping in unit tests

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

[Makes the do_afters list behave as expected](https://github.com/tgstation/tgstation/commit/b7cd6c520f25bc9570edc1e478123bf69884eb6d)

Half the code expected it to be a flat list of things we are acting on, half expected it to be an assoc list of thing -> interaction count

Let's make that second dream a reality yeah?

[Disables stoplag's lag stoppage for unit tests](https://github.com/tgstation/tgstation/commit/88b2bb371cba4a0b2b042b866cb0d1f00453487c)

Because of the unique environment of unit tests (High cpu usage, lots of sleeping) this risks spurious hard deletes from neverending stoplag runs.

## Why It's Good For The Game

Less errors/guesswork in testing, better code
Potentially addresses #77865

### CODE REVIEW

File: code/__HELPERS/stoplag.dm [hunk 1]
```diff
@@ -10,12 +10,19 @@
 		return 1
 	if (!initial_delay)
 		initial_delay = world.tick_lag
+// Unit tests are not the normal environemnt. The mc can get absolutely thigh crushed, and sleeping procs running for ages is much more common
+// We don't want spurious hard deletes off this, so let's only sleep for the requested period of time here yeah?
+#ifdef UNIT_TESTS
+	sleep(initial_delay)
+	return CEILING(DS2TICKS(intial_delay), 1)
```
Comment (LemonInTheDark) [inline code/__HELPERS/stoplag.dm:17]:
```suggestion
	return CEILING(DS2TICKS(initial_delay), 1)
```

File: code/__HELPERS/mobs.dm [hunk 2]
```diff
@@ -320,9 +320,21 @@ GLOBAL_LIST_EMPTY(species_list)
 		progbar.end_progress()
 
 	if(interaction_key)
+		var/reduced_interaction_count = LAZYACCESS(user.do_afters, interaction_key) - 1 || 0
```
Comment (Mothblocks) [inline code/__HELPERS/mobs.dm:323]: This is going to error because you're going to get `null - 1`.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__HELPERS/mobs.dm (hunk 2 above)
Comment (Mothblocks) [reply code/__HELPERS/mobs.dm:323]: Nvm null - 1 is -1 that's so silly

In that case no need for || 0

Comment (LemonInTheDark) [inline code/__HELPERS/mobs.dm:323]:
```suggestion
		var/reduced_interaction_count = (LAZYACCESS(user.do_afters, interaction_key) || 0) - 1
```
Let's just do this

**Pull request merged**